swift/apple/FirezoneKit/Sources/FirezoneKit/Models/FirezoneId.swift

```diff
@@ -3,130 +3,66 @@
-      kSecValueData: uuid.toData()
-    ]) { (_, new) in new }
-
-    try Keychain.add(query: query)
+  private enum Error: Swift.Error {
+    case SharedFolderNotAvailable
   }
 
-  // Attempt to load the firezone-id from the Keychain
-  public static func load() throws -> FirezoneId? {
-    guard let idRef = Keychain.search(query: query)
-    else { return nil }
-
-    guard let data = Keychain.load(persistentRef: idRef)
-    else { return nil }
-
-    guard data.count == UUID.sizeInBytes
+  public static func save(_ id: String) {
+    guard let fileURL = FileManager.default.containerURL(
+      forSecurityApplicationGroupIdentifier: BundleHelper.appGroupId)?
+      .appendingPathComponent("firezone-id")
     else {
-      fatalError("Firezone ID loaded from keychain must be exactly \(UUID.sizeInBytes) bytes")
+      // Nothing we can do about disk errors
+      return
     }
 
-    let uuid = UUID(fromData: data)
-    return FirezoneId(uuid)
+    try? id.write(
+      to: fileURL,
+      atomically: true,
+      encoding: .utf8
+    )
   }
 
-  // Prior to 1.4.0, our firezone-id was saved in a file. Starting with 1.4.0,
-  // the macOS client uses a system extension, which makes sharing folders with
-  // the app cumbersome, so we moved to using the keychain for this due to its
-  // better ergonomics. If the old firezone-id doesn't exist, this function
-  // is a no-op.
-  //
-  // Can be refactored to remove the file check once all clients >= 1.4.0
-  public static func migrate() throws {
-    guard try load() == nil
-    else { return } // New firezone-id already saved in Keychain
-
+  public static func load(_ version: Version) -> String? {
+    let appGroupId = switch version {
+    case .Post_1_4_0:
+      BundleHelper.appGroupId
+    case .Pre_1_4_0:
 #if os(macOS)
-    let appGroupIdPre_1_4_0 = "47R2M6779T.group.dev.firezone.firezone"
+      "47R2M6779T.group.dev.firezone.firezone"
 #elseif os(iOS)
-    let appGroupIdPre_1_4_0 = "group.dev.firezone.firezone"
+      "group.dev.firezone.firezone"
 #endif
+    }
 
-    guard let containerURL = FileManager.default.containerURL(
-      forSecurityApplicationGroupIdentifier: appGroupIdPre_1_4_0)
-    else { fatalError("Couldn't find app group container") }
-
-    let idFileURL = containerURL.appendingPathComponent("firezone-id")
-
-    // If the file isn't there or can't be read, bail
-    guard FileManager.default.fileExists(atPath: idFileURL.path),
-          let uuidString = try? String(contentsOf: idFileURL)
-    else { return }
-
-    let firezoneId = FirezoneId(UUID(uuidString: uuidString))
-    try firezoneId.save()
-  }
-
-  public static func createIfMissing() throws -> FirezoneId {
-    guard let id = try load()
+    guard let containerURL =
+            FileManager.default.containerURL(
+              forSecurityApplicationGroupIdentifier: appGroupId),
+          let id =
+            try? String(
+              contentsOf: containerURL.appendingPathComponent("firezone-id"))
```

Comment:
Couldn't you do `containerURL?.appendingPathComponent` here to avoid the `guard` if you end up returning `nil` anyway?